From b9277c294f98600efe1fd5a0b10aaa9e3a767a46 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Fri, 20 Sep 2019 19:37:19 +0300
Subject: [PATCH 213/245] drivers: net: phy: aquantia: enable USX AN for
 USXGMII protocol

commit 515e55dd0c3d4e013ff6091e163fb6cfefc65f77 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux.git

Depending on FW defaults USX AN in AQR PHY must be explicitly enabled when
using USXGMII.  Enable it based on interface type.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/phy/aquantia.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/phy/aquantia.c b/drivers/net/phy/aquantia.c
index 5a9a292d07c9..c39a5ae70447 100644
--- a/drivers/net/phy/aquantia.c
+++ b/drivers/net/phy/aquantia.c
@@ -67,6 +67,9 @@
 #define AQUANTIA_VND1_GSYSCFG_5G		3
 #define AQUANTIA_VND1_GSYSCFG_10G		4
 
+#define MDIO_PHYXS_VEND_PROV2			0xC441
+#define MDIO_PHYXS_VEND_PROV2_USX_AN		BIT(3)
+
 static int aquantia_write_reg(struct phy_device *phydev, int devad,
 			      u32 regnum, u16 val)
 {
@@ -267,6 +270,10 @@ static int aquantia_config_aneg_set_prot(struct phy_device *phydev)
                               aquantia_syscfg[if_type].syscfg);
 	}
 
+	if (if_type == PHY_INTERFACE_MODE_USXGMII)
+		phy_write_mmd(phydev, MDIO_MMD_PHYXS, MDIO_PHYXS_VEND_PROV2,
+                              MDIO_PHYXS_VEND_PROV2_USX_AN);
+
         /* wake PHY back up */
         phy_write_mmd(phydev, MDIO_MMD_VEND1, AQUANTIA_VND1_GLOBAL_SC, 0);
         mdelay(10);
-- 
2.17.1

